proposals/lip-0052.md

```diff
@@ -492,378 +458,817 @@ crossChainTransferParams = {
 }
 ```
 
-##### Parameters Validity
+##### Verification
+
+```python
+def verify(trs: Transaction) -> None:
 
-The `params` property of a cross-chain NFT transfer transaction is valid if:
+    chainID = nftID[:CHAIN_ID_LENGTH]
 
-* `recipientAddress` is a byte array of length `ADDRESS_LENGTH`,
-* `tokenID.chainID` is either `CHAIN_ID_ALIAS_NATIVE` or `receivingChainID`,
+    if chainID not in [CHAIN_ID_ALIAS_NATIVE, receivingChainID]:
+        raise Exception("NFT must be native to either to the mainchain or receiving chain.")
```

Comment:
```suggestion
        raise Exception("NFT must be native to either to the sending or receiving chain.")
```

Reply:
I am not sure if we want to allow to transfer mainchain NFTs directly from sidechain to sidechain. For the moment this is not really an issue as the mainchain does not have any NFT